extensions/standard-processors/processors/GetTCP.cpp

```diff
@@ -285,21 +285,17 @@ void GetTCP::onTrigger(const std::shared_ptr<core::ProcessContext> &context, con
       } else {
         logger_->log_error("Could not create socket for %s", endpoint);
       }
-      auto* future = new std::future<int>();
       std::unique_ptr<utils::AfterExecute<int>> after_execute = std::unique_ptr<utils::AfterExecute<int>>(new SocketAfterExecute(running_, endpoint, &live_clients_, &mutex_));
       utils::Worker<int> functor(f_ex, "workers", std::move(after_execute));
-      if (client_thread_pool_.execute(std::move(functor), *future)) {
-        live_clients_[endpoint] = future;
-      }
+      auto* future = new std::future<int>(client_thread_pool_.execute(std::move(functor)));
```

Comment:
Who cleans up these futures allocated with `new`?

Reply:
It's cleaned up by subsequent `onTrigger` calls, see the branch below. The list is also modified by `SocketAfterExecute`, which looks like it leaks on error, but I didn't verify.

Reply:
Looking at the code it looks like `SocketAfterExecute::isFinished` is indeed leaking the future. I think simply storing an `std::future` in the `GetTCP::live_clients_` should clear this up.
The whole `SocketAfterExecute` class is kinda suspicious. It stores a copy of `GetTCP::running_` while storing a pointer to `GetTCP::live_clients_` and `GetTCP::mutex_`. Isn't it supposed to take a reference to `GetTCP::running_` as well?
What are the lifetime relation of these objects? Can `SocketAfterExecute` outlive `GetTCP`? If so storing an `std::weak_ptr` would be safer than raw pointers. Otherwise references should would work as well but harder to reason about. Maybe store `GetTCP::running_`, `GetTCP::live_clients_` and `GetTCP::mutex_` in a struct and create a `std::shared_ptr` to it?